### DISCUSSION TITLE
[suggestion] SQLite binding

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#290] [Ideas]
There is ODBC driver for SQLite but do we really need to use such a thing with SQLite? SQLite is a small embedded database engine. It's better if we could handle SQLite directly without having to use ODBC.

**Comment by objeck** [comment]
Good suggestion. I will look into it when I get my dev workstation running later this week.

**Reply by objeck** [reply]
Ok, I will pick this up next. It does not seem too complex; however, I may create a separate project for this effort.

**Reply by objeck** [reply]
Not sure what to do here. The existing ``Database.ODBC`` bundle could wrap SQLite C++ calls or just use an SQLite ODBC driver.

**Reply by ghost** [reply]
Handle SQLite directly without ODBC. We don't need ODBC for something like SQLite.

**Reply by ghost** [reply]
@objeck Is your development workstation up and running yet? Or are you still working on your laptop?

**Comment by objeck** [comment]
I am mostly up and running

**Comment by objeck** [comment]
Why create an SQLite driver if using the SQLite ODBC driver (for Windows, Linux, and macOS) works fine? This framework to write a direct C++ SQLite interface will look nearly identical to the existing C++ ODBC wrapper.

**Reply by ghost** [reply]
Most languages have SQLite bindings to work directly with SQLite databases without having to use ODBC. But if you think it is duplicated with what ODBC already provides and not worth the effort, then feel free to ignore this suggestion.

**Comment by ghost** [comment]
It seems you decided to improve the ODBC support instead. I respect your choice.
